### ISSUE TITLE
Voting config exclusions APIs return \n response

### ISSUE THREAD

**Issue by russcam** [elastic/elasticsearch#57639]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.8.0-SNAPSHOT

**Plugins installed**: []

**JVM version** (`java -version`): bundled JDK

**OS version** (`uname -a` if on a Unix-like system): Windows 10 

**Description of the problem including expected versus actual behavior**:

The Voting Config Exclusions APIs (POST and DELETE) both return `\n` as the response with a content-type header of `application/json` and a content-length of 1. This response isn't valid JSON, and may require special handling in a language client, for example,

https://github.com/elastic/elasticsearch-net/pull/4738/files#diff-b2b266a90c4f392e9d643a396ef6a2d2R28-R38

**Steps to reproduce**:

 1. Make a POST to `/_cluster/voting_config_exclusions?node_names=node1%2Cnode2` or a DELETE request to `/_cluster/voting_config_exclusions?wait_for_removal=true`
 2. Observe a response similar to

```
HTTP/1.1 200 OK
content-type: application/json; charset=UTF-8
content-length: 1



```

### Possible solution(s)

- Return no response body

or

- Return an empty JSON object `{}`

**Comment by DaveCTurner** [comment]
:+1: The empty string (mod whitespace) is not valid JSON. I would prefer that we return an empty JSON object but either solution works for me, and the actual fix is rather simple. The bigger question is why the REST test suite didn't already notice this, which is why I've labelled this for the attention of the core/infra team. I investigated a bit myself but couldn't immediately identify a good place to catch this.

**Comment by russcam** [comment]
Would another possible solution here be to return an acknowledged response i.e.

```json
{ "acknowledged": true }
```
?

For the YAML tests, I think a match on `$body` for the expected response could be added to make the assertion. For example

https://github.com/elastic/elasticsearch/blob/3a15d84af92fe9b25b9dd7e3621804996a59843d/rest-api-spec/src/main/resources/rest-api-spec/test/cluster.voting_config_exclusions/10_basic.yml#L14-L17

could become

```yaml
  - do:
      cluster.post_voting_config_exclusions:
        node_ids: nodeId

  - match: { $body: {} }
```

or if an acknowledged response

```yaml
  - do:
      cluster.post_voting_config_exclusions:
        node_ids: nodeId

  - is_true: acknowledged
```

**Comment by DaveCTurner** [comment]
No, I don't think an acknowledged response is appropriate here. `acknowledged: true` means that every node in the cluster has applied the resulting cluster state, but that's irrelevant for voting config exclusions since these things are only used on the master.

**Comment by DaveCTurner** [comment]
> I'm looking for a way to assert more broadly that we never ... return something that isn't valid JSON

My attempts to do this timed out and slipped down my list. I don't think I will return to this task for the foreseeable future.
